server/app/repository/ApplicationRepository.java

```diff
@@ -160,19 +159,16 @@ private ApplicationModel submitApplicationInternal(
         app.setLifecycleStage(LifecycleStage.OBSOLETE);
         app.save();
       }
-      application.setEligibilityDetermination(eligibilityDetermination);
-      application.setApplicantData(applicant.getApplicantData());
-      application.setLifecycleStage(LifecycleStage.ACTIVE);
-      application.setSubmitTimeToNow();
-      if (tiSubmitterEmail.isPresent()) {
-        application.setSubmitterEmail(tiSubmitterEmail.get());
-      }
+      application
+          .setEligibilityDetermination(eligibilityDetermination)
+          .setApplicantData(applicant.getApplicantData())
+          .setLifecycleStage(LifecycleStage.ACTIVE)
+          .setSubmitTimeToNow();
+      tiSubmitterEmail.ifPresent(application::setSubmitterEmail);
       application.save();
 
       transaction.commit();
       return application;
-    } finally {
```

Comment:
question (non-blocking): Is `transaction.end` called automatically now because the transition is passed in as a param to `try`?